### PR TITLE
Support Whitelisted resources

### PR DESCRIPTION
**Pull request by russokj**
Problem: To have feature parity with the original OpenShift
Big-IP router, we need to allow the user the ability to create
resources and then have the controller manage them.
    
Solution: CCCL will keep track of changes made by the CCs to
whitelisted resources.  These changes need to be backed out 
when new changes are requested by CC for the resource.

### CODE REVIEW

**Comment by edarzins** [comment]
So, it's up to the controller to determine which BIG-IP resources are white-listed, and that info is passed to CCCL?

**Comment by russokj** [comment]
The controller does not know what is white listed or not.  The user will have to add metadata to specific resources that they want to have control over.

File: f5_cccl/resource/resource.py [hunk 1]
```diff
@@ -25,10 +25,28 @@
...
+        val = "NOT DICT: CMP DST {} to SRC {}".format(dst, src)
+        LOGGER.error(val)
+        return src
+
+    for key in src.keys():
+        dst[key] = merge_dict(dst[key], src[key]) if key in dst else src[key]
+        val = "MERGED TO DST[{}]: {}".format(key, dst[key])
+        LOGGER.error(val)
+
+    LOGGER.error("RETURNED DST")
```
Comment (edarzins) [inline f5_cccl/resource/resource.py]: The LOGGER calls in this function look like your personal debug and not really errors, remove.

Comment (russokj) [reply f5_cccl/resource/resource.py]: Actually, all three log statements were debug.  Normally I put my initials in the log message so I can easily grep for them ... and this is what happens when I forget to do that!

File: setup_requirements.txt [hunk 2]
```diff
@@ -7,4 +7,6 @@ PyJWT==1.4.0
 PyYAML==3.12
 requests==2.9.1
 simplejson==3.10.0
+jsonpatch==1.21
```
Comment (edarzins) [inline setup_requirements.txt]: Have you submitted these new dependencies for license approval?

Comment (russokj) [reply setup_requirements.txt]: Yes.

**Comment by russokj** [comment]
Good news is none of the patching will occur for traditional deployments (any resource that isn't white listed).  All the nastiness is isolated to the resource merge() method.

**Pull request merged**